ethdb/database.go

```diff
@@ -90,6 +90,14 @@ type AncientReader interface {
 	AncientSize(kind string) (uint64, error)
 }
 
+// AncientBatchReader is the interface for 'batched' or 'atomic' reading.
+type AncientBatchReader interface {
+	AncientReader
+	// AtomicReadAncients runs the given read operation while ensuring that no writes take place
+	// on the underlying freezer.
+	AtomicReadAncients(fn func(AncientReader) error) (err error)
```

Comment:
Let's call this `ReadAncients`, and let's call `ReadAncients` something else, like `ReadSequential`

Reply:
or `AncientRange`